client/components/tinymce/plugins/wpcom-view/wpvideo-view.jsx

```diff
@@ -0,0 +1,140 @@
+ */
+import shortcodeUtils from 'lib/shortcode';
+import { getVideo } from 'state/videos/selectors';
+import QueryVideo from 'components/data/query-video';
+
+class WpVideoView extends Component {
+
+	static match( content ) {
+		const match = shortcodeUtils.next( 'wpvideo', content );
+
+		if ( match ) {
+			return {
+				index: match.index,
+				content: match.content,
+				options: {
+					shortcode: match.shortcode
+				}
+			};
+		}
+	}
+
+	static serialize( content ) {
+		return encodeURIComponent( content );
+	}
+
+	constructor( props ) {
+		super( props );
+	}
+
+	constrainVideoDimensions( shortcodeWidthAttribute, shortcodeHeightAttribute ) {
+		const defaultWidth = 640;
+		const defaultAspectRatio = 16 / 9;
+		let { width, height } = this.props.video ? this.props.video : { width: defaultWidth, height: defaultWidth / defaultAspectRatio };
+		const aspectRatio = width / height;
+
+		if ( shortcodeWidthAttribute && ! shortcodeHeightAttribute ) {
+			width = shortcodeWidthAttribute;
+			height = shortcodeWidthAttribute / aspectRatio;
+		} else if ( ! shortcodeWidthAttribute && shortcodeHeightAttribute ) {
+			width = shortcodeHeightAttribute * aspectRatio;
+			height = shortcodeHeightAttribute;
+		} else if ( shortcodeWidthAttribute && shortcodeHeightAttribute ) {
+			const definedAspectRatio = shortcodeWidthAttribute / shortcodeHeightAttribute;
+			if ( definedAspectRatio > aspectRatio ) {
+				width = shortcodeHeightAttribute * aspectRatio;
+				height = shortcodeHeightAttribute;
+			} else {
+				width = shortcodeWidthAttribute;
+				height = shortcodeWidthAttribute / aspectRatio;
+			}
+		}
+
+		return { width, height };
+	}
+
+	getVideoAttributes() {
+		const shortcode = shortcodeUtils.parse( this.props.content );
+		const namedAttrs = shortcode.attrs.named;
+
+		const videoDimensions = this.constrainVideoDimensions( parseInt( namedAttrs.w, 10 ), parseInt( namedAttrs.h, 10 ) );
+
+		const shortcodeAttributes = {
+			guid: shortcode.attrs.numeric[0],
+			w: videoDimensions.width,
+			h: videoDimensions.height,
+			autoplay: namedAttrs.autoplay === 'true',
+			hd: namedAttrs.hd === 'true',
+			loop: namedAttrs.loop === 'true',
+			at: parseInt( namedAttrs.at, 10 ) || 0,
+			defaultLangCode: namedAttrs.defaultlangcode
+		};
+
+		return Object.assign( {}, pick( shortcodeAttributes, ['guid', 'w', 'h'] ), { embedUrl: this.getEmbedUrl( shortcodeAttributes ) } );
+	}
+
+	getEmbedUrl( shortcodeAttributes ) {
+		const defaultAttributeValues = { hd: false, at: 0, defaultLangCode: undefined };
+		const attributesWithNonDefaultValues = omit( shortcodeAttributes, ( value, key ) => defaultAttributeValues[key] === value );
+		const queryString = QueryString.stringify( pick( attributesWithNonDefaultValues, ['autoplay', 'hd', 'loop', 'at', 'defaultLangCode'] ) );
+
+		return `https://videopress.com/embed/${ shortcodeAttributes.guid }?${ queryString }`;
+	}
+
+	onLoad() {
+		const doc = tinymce.activeEditor.iframeElement.contentDocument;
```

Comment:
I really wish we weren't pull the active editor from the global `tinymce` instance, though understandably it's not otherwise made available to this component. Can you do some digging to see if there's a way we can more explicitly pass the editor instance or iframe to the component?